libcxx/include/shared_recursive_mutex

```diff
@@ -0,0 +1,61 @@
+#ifndef _SHARED_RECURSIVE_MUTEX_
+#define _SHARED_RECURSIVE_MUTEX_
+
+#include <cstdint>
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include <shared_mutex>
+
+namespace std {
```

Comment:
This line is confusing. Do we want to put this non-standard extension into `std`? Should we use an __uglified class name (`__shared_recursive_mutex`)?

Reply:
done